x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -212,22 +260,37 @@ export class SearchSessionService
     );
   };
 
-  public extend(deps: SearchSessionDependencies, sessionId: string, expires: Date) {
+  public async extend(
+    deps: SearchSessionDependencies,
+    user: AuthenticatedUser | null,
+    sessionId: string,
+    expires: Date
+  ) {
     this.logger.debug(`extend | ${sessionId}`);
-
-    return this.update(deps, sessionId, { expires: expires.toISOString() });
+    await this.get(deps, user, sessionId); // Verify correct user
+    return this.update(deps, user, sessionId, { expires: expires.toISOString() });
   }
 
-  // TODO: Throw an error if this session doesn't belong to this user
-  public cancel = (deps: SearchSessionDependencies, sessionId: string) => {
-    return this.update(deps, sessionId, {
+  public cancel = async (
+    deps: SearchSessionDependencies,
+    user: AuthenticatedUser | null,
+    sessionId: string
+  ) => {
+    this.logger.debug(`delete | ${sessionId}`);
+    await this.get(deps, user, sessionId); // Verify correct user
```

Comment:
**question** doesn't the call to `update` also verify the current user? Is this `get` necessary?

Reply:
:+1: I've removed from the `cancel` and `extend` calls